compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrConverter.kt

```diff
@@ -95,14 +95,21 @@ class Fir2IrConverter(
         return irClass
     }
 
+    // Sort declarations so that all non-synthetic declarations are before synthetic ones.
+    // This is needed because converting synthetic fields for implementation delegation needs to know
+    // existing declarations in the class to avoid adding redundant delegated members.
+    private fun putSyntheticLast(declarations: List<FirDeclaration>) : Iterable<FirDeclaration> {
```

Comment:
I don't like name, `put` is associated with adding something into container. Consider something like `deprioritizeSynthetics` or `sortedBySynthetic`